lib/puppet/resource/capability_finder.rb

```diff
@@ -95,8 +81,40 @@ def self.query_puppetdb(query)
       end
       JSON.parse(response.body)
     end
+
+    # The format of the response body is documented at
+    #   http://docs.puppetlabs.com/puppetdb/3.0/api/query/v4/resources.html#response-format
+    unless result.is_a?(Array)
+      raise Puppet::DevError,
+      "Unexpected response from PuppetDB when looking up #{cap}: " +
+        "expected an Array but got #{result.inspect}"
+    end
+
+    result
   rescue JSON::JSONError => e
```

Comment:
top level rescue sometimes cause problems (have seen it in testing/mocking) - better to always use a begin/end